lib/events.js

```diff
@@ -153,7 +153,7 @@ api.addBatch = async ({
         throw result.error;
       }
     } else if(jsonld.hasValue(event, 'type', 'ContinuityMergeEvent') &&
-      _temp.insideBatch.length !== 0) {
+      deferredEvents.length > 0 && _temp.insideBatch.length !== 0) {
```

Comment:
Seems like we should use `_temp.insideBatch.length > 0` for consistency with the other length check.

That said, can you explain why this is ok to skip when there are no deferred events? Is this saying that if there are no deferred events then the merge event can't possibly be a descendant of anything that was deferred, so there's no reason to consider deferring it as well? If so, and the ordering that events are visited here ensures that merge events are always visited *after* their ancestors, then this check seems fine to me. I'm assuming the order is *trusted* (we aren't relying on an untrusted peer's ordering of the events we're processing, but rather, we've made sure it's legit).

Reply:
Yes, you have the right idea.  Ordering of gossip is currently occurring on the gossip target side of things using the kahn's algo.  Since we're are validating the DAG, we know that everything is in proper order.  So there is really not a trust issue here, it's just a matter of defining the protocol.  If the peer did not order things properly, the gossip session will be aborted and all the disincentives for doing things incorrectly on the network apply (back-off etc). 

As part of this PR, we are now hashing all the events as a preliminary step.  So it would be easy to add a sort on the receiving side of things instead of or in addition to the target peer side.  This needs further consideration and if we decide to do something differently, it would need to be in a separate PR.